src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -13,39 +13,37 @@
 import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.math.system.plant.DCMotor;
 import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.trajectory.TrapezoidProfile.State;
 import edu.wpi.first.units.Distance;
 import edu.wpi.first.units.Measure;
 import edu.wpi.first.wpilibj.simulation.ElevatorSim;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.InstantCommand;
-import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.TrapezoidProfileSubsystem;
 import frc.robot.Constants.ElevatorConstants;
 import monologue.Annotations.Log;
 import monologue.Logged;
 
 // TODO change to inherit from TrapezoidProfileSubsystem. This will have an impact on how the
 // simulation is done.
-public class Elevator extends SubsystemBase implements Logged {
-
-  // TODO remove the isReal member
-  private boolean isReal;
+public class Elevator extends TrapezoidProfileSubsystem implements Logged {
 
   // TODO make all of these member variables final
-  private CANSparkMax leftMotor;
-  private CANSparkMax rightMotor;
+  private final CANSparkMax leftMotor;
+  private final CANSparkMax rightMotor;
 
-  private ElevatorSim sim;
+  private final AbsoluteEncoder encoder;
 
-  private AbsoluteEncoder encoder;
+  private final SparkPIDController feedback;
 
-  private SparkPIDController realFeedback;
+  private final ElevatorFeedforward feedforward;
+
+  private ElevatorSim sim;
```

Comment:
@JakeStew81  The sim objects should also be final. When you change these to final, code won't compile because these don't get initialized in the constructor. To fix that, add an else block to the isReal check. Explicitly set these to null. 

What you have currently, and this change are functionally equivalent. If the robot is real, these objects are null. This pattern makes object creation consistent for all objects whether they are real or sim. It also explicitly calls out to the programmer when these objects are null and when they shouldn't be used.